### DISCUSSION TITLE
Using same seed phrase, can we derive Eth address on Aptos?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#73] [Questions] [answered]
Asked by `aasoma` in discord
> Hey guys, I wanted to understand how addresses are encoded in Aptos. Recently a friend of mine sent APT from an exchange to an Aptos address but instead of inputting her Aptos address she instead input her ERC20 address. Shockingly, the transaction went through and the funds were transferred to an Aptos recieving address but the recieving address is the same as her ERC20 address.
Is there a way to access this Aptos wallet address with the same private key as ERC wallet 
Sharing Aptos TX hash if someone can help

**Comment by gregnazario** [comment, marked as answer]
Addresses are a hash of a public key, so it would be nearly impossible to get the same address.

On Ethereum, your address is your public key.  So, tl;dr please check that you have an address on Aptos that is not your EVM address, but instead an Aptos address with an Aptos wallet.

Please see this document on how addresses are created: https://aptos.dev/concepts/accounts/#authentication-key

**Reply by aahanasomani** [reply]
Hey @gregnazario thanks for the prompt response. Looks like the wallet is then inaccessible via the old Ethereum private key.
Is there any way the user can get the funds back. Have heard so many people on Discord, Telegram, Twitter and even Reddit who have lost substantial funds because they put an ERC20 address accidentally and Aptos accepted it creating a new wallet that is inaccessible now. (Cumulatively my estimate is that this could be 6 to 7 digit dollar figure in losses for the affected users). These funds are just sitting idle in inaccessible wallets resulting in such a huge loss of economic activity (also we should not undermine the pain of people who were affected because of the way our specific Blockchain address schema is coded)

Wondering if it is possible to get these funds back to all these affected users via a network upgrade of some sort in future. Would love to know your thoughts.

**Reply by aahanasomani** [reply]
Hey @gregnazario 
On Aptos network, if someone sends funds to an Ethereum Address, it appears as if Aptos is creating a new Aptos wallet address for him. Who controls the private key of this wallet?

**Reply by gregnazario** [reply]
Yeah, I totally agree it is problematic to send funds to a non-existent account.

In theory, it could be possible to have a blockchain upgrade to have EVM compatible addresses, but that does bring some possible attack vectors associated, where public keys match other addresses.

In theory the private key of the wallet, could be anything.  Don't know what the key would be, since it's a SHA-256 hash (basically full-randomization) of the public key.  It's computationally near impossible to get the private key with the different address scheme.

**Reply by aahanasomani** [reply]
Got it. I am trying to understand the rationale behind such an implementation in the first place. If a user inputs an address that is non existent, why are we allowing such transfers in the first place? Shouldn't be the first call of action here to tell the user that this is a non-existent wallet and or it doesn't match the address schema on Aptos. Instead we are creating a new public key deriving it from his wrong address and allowing these erroneous fund transfers. I am guessing there must be a strong reason for Aptos to have such an architecture in the first place. Any ideas on what that might be?

**Reply by aahanasomani** [reply]
Hey @gregnazario 
Waiting for your reply!

**Reply by gregnazario** [reply]
Most wallets will tell you that the wallet doesn't exist, and that it will be created.

It depends on the platform you're using to transfer the tokens.  In the case of most exchanges, they usually skip this kind of information.

It's no different than on Eth and other locations as well.

**Reply by gregnazario** [reply]
Here's an example from Petra wallet that verifies that the account exists to be sent to.
![Screenshot 2024-03-22 at 9 48 51 AM](https://github.com/aptos-labs/aptos-developer-discussions/assets/1335270/01463d06-534a-4d29-a0f5-414fd05e98ad)
![Screenshot 2024-03-22 at 9 49 23 AM](https://github.com/aptos-labs/aptos-developer-discussions/assets/1335270/0d7e0c44-3504-4c29-af7b-d3d7dce2d65c)

**Comment by aahanasomani** [comment]
Hey guys,
Any update on this? Do we know who controls the private keys to this newly created wallets??

**Reply by 0x-j** [reply]
Hey, since it's impossible to get private key from address, so we can never find out unfortunately
